chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -440,10 +440,15 @@ class Bundle extends Record {
   /** Returns a field's contained user-defined Bundle element if it appears to
     * be one, otherwise returns None.
     */
-  private def getBundleField(m: java.lang.reflect.Method): Option[Data] = m.invoke(this) match {
-    case d: Data => Some(d)
-    case Some(d: Data) => Some(d)
-    case _ => None
+  private def getBundleField(m: java.lang.reflect.Method): Option[Data] = {
+    if (m.getDeclaringClass.isAssignableFrom(classOf[Bundle])) None
```

Comment:
Yeah, so it solves the problem (within the limits of the current architecture) at the root, and so the fix also applies for Module reflection.